src/show.jl

```diff
@@ -0,0 +1,164 @@
+function Base.show(io::IO, fid::File)
+    if isvalid(fid)
+        print(io, "HDF5 data file: ", fid.filename)
+    else
+        print(io, "HDF5 data file (closed): ", fid.filename)
+    end
+end
+
+function Base.show(io::IO, g::Group)
+    if isvalid(g)
+        print(io, "HDF5 group: ", name(g), " (file: ", g.file.filename, ")")
+    else
+        print(io, "HDF5 group (invalid)")
```

Comment:
Should we change these to `HDF5 Group` or `HDF5.Group`  (capitalize them to match the struct name?

Reply:
`HDF5.Group` and similar sounds good to me.